### PR TITLE
dashboard uses ViewComponents

### LINKED ISSUE #2045: dashboard should use ViewComponents instead of helper methods
This is for rendering the tables and such in the _information_ views.

The status views are not settled into what they are yet.

### PR DESCRIPTION
**Pull request by ndushay**
## Why was this change made? 🤔

ViewComponents make the code more testable and easier to work with

closes #2045

## How was this change tested? 🤨

⚡ ⚠ If this change has cross service impact, or if it changes code used internally for cloud replication, ***run [integration test preassembly_image_accessioning_spec.rb](https://github.com/sul-dlss/infrastructure-integration-test/blob/main/spec/features/preassembly_image_accessioning_spec.rb) against stage as it tests preservation***, and/or test in stage environment, in addition to specs. The main classes relevant to replication are `ZipmakerJob`, `PlexerJob`, `*DeliveryJob`, `ResultsRecorderJob`, and `DruidVersionZip`; [see here for overview diagram of replication pipeline](https://github.com/sul-dlss/preservation_catalog/blob/main/app/jobs/README.md).⚡

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -4,8 +4,6 @@ source 'https://rubygems.org'
 gem 'aws-sdk-s3', '~> 1.17'
 gem 'committee' # Validates HTTP requests/responses per OpenAPI specification
 gem 'config' # Settings to manage configs on different instances
-gem 'dor-event-client', '~> 1.0'
-gem 'dor-workflow-client', '~> 5.0' # audit errors are reported to the workflow service
```
Comment (ndushay) [inline Gemfile:8]: these gems have been listed in the wrong place for a while;  I just noticed.

File: app/components/dashboard/complete_moabs_component.html.erb [hunk 2]
```diff
@@ -19,11 +18,11 @@
           <td><%= num_preserved_objects %></td>
           <td><%= complete_moab_total_size %></td>
           <td><%= complete_moab_average_size %></td>
-          <td><%= complete_moab_status_counts[0] %>
+          <td><%= complete_moab_status_counts[0] %></td>
```
Comment (ndushay) [inline app/components/dashboard/complete_moabs_component.html.erb:21]: fixing a whoops

File: app/components/dashboard/audit_information_component.html.erb [hunk 3]
```diff
@@ -19,7 +18,7 @@ TBD:  make the audit check description into a tooltip?
       <td>validate_moab</td>
       <td><strong>on demand</strong></td>
       <td><strong>one</strong>:<br/>queued by preservation robot step after updating a Moab</td>
-      <td class="text-end<%= ' table-danger' if moab_audits_older_than_threshold? %>">Moab audits older than <%= moab_audit_age_threshold %>:<hr/><%= num_moab_audits_older_than_threshold %></td>
+      <td class="text-end<%= ' table-warning' if moab_audits_older_than_threshold? %>">Moab audits older than <%= moab_audit_age_threshold %>:<hr/><%= num_moab_audits_older_than_threshold %></td>
```
Comment (ndushay) [inline app/components/dashboard/audit_information_component.html.erb:21]: turns out the bootstrap class is `table-warning` ... whoops

File: app/components/dashboard/moab_storage_roots_component.html.erb [hunk 4]
```diff
@@ -28,23 +27,27 @@
             <td><%= info[2] %></td><%# average size %>
             <td class="text-end"><%= info[3] %></td><%# moab count %>
             <td class="text-end"><%= info[4] %></td><%# ok status %>
-            <% info[5..].map do |val| %>
+            <% info[5..-2].map do |val| %>
```
Comment (ndushay) [inline app/components/dashboard/moab_storage_roots_component.html.erb:30]: this whole business with the storage_root_info structure is nasty and I made a ticket to fix it

File: app/components/dashboard/moab_storage_roots_component.html.erb [hunk 5]
```diff
@@ -28,23 +27,27 @@
...
               <td class="text-end<%= ' table-danger' if val != 0 %>"><%= val %></td>
             <% end %>
+            <% num_expired_checksum = info[10] %>
```
Comment (ndushay) [inline app/components/dashboard/moab_storage_roots_component.html.erb:33]: this one should be warning, not danger, styling

File: app/components/dashboard/zip_parts_suffixes_component.html.erb [hunk 6]
```diff
@@ -12,7 +12,7 @@
       <tbody>
         <tr>
           <td>.zip</td>
-          <td class="text-end">0<%= zip_part_suffixes['.zip'] %></td>
+          <td class="text-end"><%= zip_part_suffixes['.zip'] %></td>
```
Comment (ndushay) [inline app/components/dashboard/zip_parts_suffixes_component.html.erb:15]: fixing a typo

File: app/helpers/dashboard_audit_helper.rb [hunk 7]
```diff
@@ -1,62 +1,45 @@
...
 # helper methods for dashboard pertaining to audit functionality
 module DashboardAuditHelper
-  # CompleteMoab.last_version_audit is the most recent of 3 separate audits:
-  #   moab_to_catalog - all CompleteMoabs are queued for this on the 1st of the month
-  #   catalog_to_moab - all CompleteMoabs are queued for this on the 15th of the month
-  #   checksum_validation - CompleteMoabs with expired checksums are queued weekly;  they expire after 90 days
-  # 18 days gives a little slop for either of the first 2 audit queues to die down.
-  MOAB_LAST_VERSION_AUDIT_THRESHOLD = 18.days
-
-  REPLICATION_AUDIT_THRESHOLD = 90.days # meant to be the same as PreservationPolicy.archive_ttl
```
Comment (ndushay) [inline app/helpers/dashboard_audit_helper.rb:12]: this is now in the service class

File: app/helpers/dashboard_audit_helper.rb [hunk 8]
```diff
@@ -1,62 +1,45 @@
...
-    (DateTime.now - REPLICATION_AUDIT_THRESHOLD).to_s
-  end
-
-  def num_replication_audits_older_than_threshold
-    PreservedObject.archive_check_expired.count
-  end
-
-  def replication_audits_older_than_threshold?
-    num_replication_audits_older_than_threshold.positive?
-  end
```
Comment (ndushay) [inline app/helpers/dashboard_audit_helper.rb:61]: all this is in service class;  it is only used by the new view components.

File: app/helpers/dashboard_catalog_helper.rb [hunk 9]
```diff
@@ -1,123 +1,44 @@
...
+
 # helper methods for dashboard pertaining to catalog functionality
 module DashboardCatalogHelper
+  # used by catalog_status partials
   def catalog_ok?
     num_preserved_objects == num_complete_moabs &&
       num_object_versions_per_preserved_object == num_object_versions_per_complete_moab
   end
 
-  def storage_root_info # rubocop:disable Metrics/AbcSize
```
Comment (ndushay) [inline app/helpers/dashboard_catalog_helper.rb:10]: anything dropped here is in the new service class and is only used by the new ViewComponents

File: app/helpers/dashboard_replication_helper.rb [hunk 10]
```diff
@@ -21,14 +25,7 @@ def endpoint_data
     endpoint_data
   end
 
-  def zip_part_suffixes
```
Comment (ndushay) [inline app/helpers/dashboard_replication_helper.rb:24]: ditto:  in service class, only used by view components

**Review by ndushay — COMMENTED** [review]
Main points:

- use ViewComponents 
- using service classes instead of helpers.  (some helper methods remain until we settle on our preferred compact status display)

File: app/components/dashboard/audit_information_component.html.erb [hunk 11]
```diff
@@ -46,16 +45,16 @@ TBD:  make the audit check description into a tooltip?
       <td>checksum_validation</td>
       <td><strong>Weekly</strong>: 1am on Sunday<br/><em>for CompleteMoabs with expired checks</em></td>
       <td><strong>subset</strong>:<br/>expired checksums queued by associated MoabStorageRoot</td>
-      <td class="text-end<%= ' table-danger' if moab_audits_older_than_threshold? %>">Moab audits older than <%= moab_audit_age_threshold %>:<hr/><%= num_moab_audits_older_than_threshold %></td>
+      <td class="text-end<%= ' table-warning' if moab_audits_older_than_threshold? %>">Moab audits older than <%= moab_audit_age_threshold %>:<hr/><%= num_moab_audits_older_than_threshold %></td>
       <td class="text-end<%= ' table-danger' if CompleteMoab.invalid_checksum.count != 0 %>"><%= CompleteMoab.invalid_checksum.count %></td>
-      <td class="text-center"><%= num_expired_checksum_validation %><br/><br/>(passing checks expire 90 days after they are run)</td>
+      <td class="text-center<%= ' table-warning' if num_expired_checksum_validation != 0 %>"><%= num_expired_checksum_validation %><br/><br/>(passing checks expire 90 days after they are run)</td>
```
Comment (justinlittman) [inline app/components/dashboard/audit_information_component.html.erb:50]: I'd encourage moving the logic into the ruby part of the component here and elsewhere.

File: app/components/dashboard/moab_storage_roots_component.rb [hunk 12]
```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+module Dashboard
+  # methods for dashboard pertaining to MoabStorageRootsComponent
+  class MoabStorageRootsComponent < ViewComponent::Base
+    attr_reader :dashboard_catalog_service
+
+    delegate :status_labels,
```
Comment (justinlittman) [inline app/components/dashboard/moab_storage_roots_component.rb:8]: I wonder if it wouldn't be easier/cleaner to turn each of these services into a mixin.

**Review by justinlittman — APPROVED** [review]
Approve pending consideration of comments.

**Comment by ndushay** [comment]
I'm merging this and have made tickets #2078 and #2079 to address JLitt's concerns;  I have assigned those tickets to myself and am beginning work immediately

**Pull request merged**